airbyte-integrations/connectors/destination-s3-v2/src/main/kotlin/S3V2Writer.kt

```diff
@@ -40,20 +47,41 @@ class S3V2Writer(
 
     @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION", justification = "Kotlin async continuation")
     inner class S3V2StreamLoader(override val stream: DestinationStream) : StreamLoader {
-        private val partNumber = AtomicLong(0L) // TODO: Get from destination state
+        private val partNumber = AtomicLong(0L)
+
+        override suspend fun start() {
+            val state = destinationStateManager.getState(stream)
+            val maxPartNumber =
+                state.generations
+                    .map {
+                        println(it)
```

Comment:
I'll remove these

Reply:
Lol, I have to, some of them are breaking docker-tests because they go to stdout and get picked up by the output consumer:
```
io.airbyte.protocol.models.v0.AirbyteMessage@222bf7e4[type=<null>,log=<null>,spec=<null>,connectionStatus=<null>,catalog=<null>,record=<null>,state=<null>,trace=<null>,control=<null>,additionalProperties={generations_by_state={FINALIZED={0={s3-v2-test/test202410305hXtC/test_stream/0.jsonl=0}}}}]
```